### PR TITLE
[ONNX] Fixed export for random

### PR DESCRIPTION
**Pull request by neginraoof**
[ONNX] Fixed export for random generator ops

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1621,20 +1621,24 @@ def _pad_packed_sequence(g, data, batch_sizes, batch_first, padding_value, total
...
-    shape = sym_help._maybe_get_const(shapes_list[0], "is")
+    shape = sym_help._get_const(shapes_list[0], "is", "rand")
     return g.op('RandomUniform', shape_i=shape)
 
 
 def randn_like(g, self, *others):
     return g.op('RandomNormalLike', self)
 
 
+def rand_like(g, self, *others):
```
Comment (lara-hdr) [inline torch/onnx/symbolic_opset9.py]: one of the args of these ops is dtype.
Can you add a Cast when necessary to make sure to match the desired type?
And maybe change one of the tests to set the dtype for a type other than float.

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1621,20 +1621,24 @@ def _pad_packed_sequence(g, data, batch_sizes, batch_first, padding_value, total
...
+    return g.op('RandomUniformLike', self)
```
Comment (lara-hdr) [inline torch/onnx/symbolic_opset9.py]: Could you also add 'float' as the default dtype when dtype is None (for supporting scripting scenarios where the dtype is not explicitly set)

**Review by lara-hdr — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset9.py (hunk 1 above)
Comment (neginraoof) [reply torch/onnx/symbolic_opset9.py]: Sure. Double was another type supported by torch.rand_like. I've added tests for that.

**Comment by rajeevsrao** [comment]
@lara-hdr can you please approve?

**Review by lara-hdr — APPROVED** [review]
LGTM

**Comment by rajeevsrao** [comment]
@neginraoof can you please merge this?

**Comment by neginraoof** [comment]
cc @houseroad this PR is approved and ready.

**Review by houseroad — APPROVED** [review]
LG

**Comment by neginraoof** [comment]
@houseroad is there a related failure?

**Comment by houseroad** [comment]
This PR was reverted since it broke CIs. Could you resubmit the PR @neginraoof ?

**Pull request closed without merging**